feat(exasol): add support for TypeOf

Nicoretti opened the pull request:
## Description of changes
Add implementation for TypeOf to Exasol backend

cpcloud commented on ibis/backends/exasol/compiler.py:
```diff
@@ -187,3 +186,6 @@ def visit_DayOfWeekName(self, op, *, arg):
     def visit_ExtractQuarter(self, op, *, arg):
         return self.cast(self.f.to_char(arg, "Q"), op.dtype)
+
+    def visit_TypeOf(self, op, *, arg):
```
Simple mappings like this can be added to `SIMPLE_OPS`.

Nicoretti replied:
Nice, thanks.

Nicoretti commented:
@cpcloud, it appears that there are a few follow-up items here that I hadn't initially noticed. I'll close this for the time being and will recreate it once I've addressed those issues.

Side note: I plan to submit some more PRs next week.
